MUST NOT store min_ack_delay (#63)
quicwg/ack-frequency: draft-ietf-quic-ack-frequency.md

the peer is allowed to send ACK_FREQUENCY frames independent of whether it also
sends the min_ack_delay transport parameter or not.

Endpoints MUST NOT remember the value of the min_ack_delay transport parameter

> they received. Due to this, ACK_FREQUENCY frames MUST NOT be sent in 0-RTT

Reviewer: ```suggestion
they received. Consequently, ACK_FREQUENCY frames MUST NOT be sent in 0-RTT
```

Reviewer 2: isn't it "cannot"?

(You see, if we say "MUST NOT", we probably need to say what happens if you get a 0-RTT packet with ACK_FREQUENCY in it, but I know that some implementations can't really make that distinction.)

Author: Good point, I think cannot is more correct, updated.

> packets.

Reviewer: ```suggestion
packets, as per Section 7.4.1 of {{QUIC-TRANSPORT}}.
```

the peer is allowed to send ACK_FREQUENCY frames independent of whether it also
sends the min_ack_delay transport parameter or not.

> An endpoint MUST NOT send ACK_FREQUENCY frames until it has received the
> min_ack_delay transport parameter from the peer.  When clients use 0-RTT, they
> MUST NOT store the value of the server's min_ack_delay transport parameter.

Reviewer 3: How about going like:
```suggestion
Endpoints MUST NOT remember the value of min_ack_delay transport parameter they received.
```

I think provision of the first sentence ("An endpoint MUST NOT send ACK_FREQUENCY frames until ...") is covered by the last sentence of the previous paragraph. If we think that that sentence is vague, we can change "if" of that sentence to "if and only if."

Regarding the second sentence, my preference goes to using the phrase "remembered" and talking about both sides, because that is what RFC 9000 does.

Author: Thanks, suggestion accepted.

> Due to this, ACK_FREQUENCY frames MUST NOT be sent in 0-RTT packets.

Reviewer 2: Please drop this last sentence.

Author: I took Kazuho's suggestion, which I think is correct based on the discussion on #22?
